### ISSUE TITLE
Package Manager: [Mac/Ubu IDE] "GMPM Failure code 126" dialog shown when downloading Prefab/GMRT packages

### ISSUE THREAD

**Issue by yerumaku** [YoYoGames/GameMaker-Bugs#8345]
### Description

I see an error message when I try to download Prefab. 

This error generally applies to all GMPM resources!

![Image](https://github.com/user-attachments/assets/a91232c0-cf83-42dc-8579-022e035be38c)

_This error started as soon as GMPM appeared._

> Installing Packages
> installing 'Release' packages
> /usr/bin/env bash -c "'/Users/yerumaku/Library/Application Support/GameMakerStudio2-Beta/yerumaku_2180665/node/node/bin/node' '/Users/yerumaku/Library/Application Support/GameMakerStudio2-Beta/yerumaku_2180665/node/node/lib/node_modules/npm/bin/npm-cli.js' install --registry https://gmpm.gamemaker.io/ --loglevel verbose"
> bash: /Users/yerumaku/Library/Application Support/GameMakerStudio2-Beta/yerumaku_2180665/node/node/bin/node: Permission denied
> potential issue when fetching packages, `npm` returned non-zero exit code
> GMPM install completed with exit code: 126

***

Additional Screen - when I drag one GMPM or scroll it, sometimes the filter display gets lost.
(see video in comments)
![Image](https://github.com/user-attachments/assets/681ac082-0ecb-4153-a0a5-b65ebaefaa52)

### Steps To Reproduce

1. Start GameMaker
2. Open project
3. Open GMPM
4. Select Prefabs filter (or other)
5. Select Prefab
6. Click Install button
7. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.1100 (Betas)

### Which operating system(s) are you seeing the problem on?

macOS Sequoia, Ubuntu 24.04

### Are you running GameMaker from inside your Steam library?

No

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by Emc1923** [comment]
Can confirm I'm having the same issue and the prefabs that were correctly installed previously won't load in the the Prefab Library.

**Comment by Emc1923** [comment]
> At what steps do you have GMPM not working on Linux? Basically, nothing loads on my macOS.

@yerumaku macOS and Linux fail at the same stage

**Comment by stuckie** [comment]
If this had been broken for you for a while, you really should have logged a ticket in so we could have investigated sooner.
It is as simple as that. 
As mentioned, this is the bug database; we do not always check discord, forums, reddit, twitter/x, mastodon, bluesky, threads, facebook, or any other social media network to try and pull bugs together. There are literally thousands here to deal with, that people have logged with us properly, without us going out of our way to find more!
In future, if you find a bug, please report it through the proper channels so that we have a good amount of time to address it.

Griping aside, we have not had the time to look into this issue yet to see what it is to be able to give you an answer, it has only been assigned for two days.
Any updates will be on this ticket and no, the Windows IDE update has nothing to do with GMPM, Plugins, or anything else.

**Comment by stuckie** [comment]
So, somewhat thanks to @gnysek on the other ticket that you started linking back to here, I realised this was a permissions issue and is now fixed.
However in future, please don't do that either - keep related discussions to issues in the ticket that has been created for it so it is easier to follow.

If you need to fix it now, you just need to set the executable bit on your node install.
Ubuntu: chmod +x /home/<YOUR_UBUNTU_USER>/.config/GameMakerStudio2-Beta/<YOUR_GM_USER>/node/node/bin/node
macOS: chmod +x /Users/<YOUR_MACOS_USER>/Library/Application\ Support/GameMakerStudio2-Beta/<YOUR_GM_USER>/node/node/bin/node

**Comment by gnysek** [comment]
@yerumaku 
> I realised this was a permissions issue **and is now fixed.**

If it's fixed, should be in next beta IMO.

Giving tons of comments, especially those not related exactly to issue isn't helping.

**Comment by gnysek** [comment]
> others may have it

Then they can report a bug about it then. Simple as that. Stuckie wrote that this is fixed, so I believe it will be OK with next release. If not - report new bug.

**Comment by stuckie** [comment]
You copied a fragment of the log, yes, but there was no full bug package to see if it was anything else, and at first glance it looks like a general file permission error as the surrounding lines do indicate that something ran - so it could have been lack of permissions to write to a directory or anything else. At that point, it was not obvious.
The other ticket showed that that is specifically an executable and that it should have the executable bit set; as it is called in a specialised manner, it is not immediately obvious that it needed that, nor is it obvious how it *lost* it in the first place.
I was then able to confirm that what you were seeing was the outcome of that by taking the executable bit *off* of my install and getting the same errors.
Thus, I was able to get the IDE to perform a check that it can run that executable, and set the executable bit if it is missing.
That was the extra piece of information that I was calling out that led to this being fixed quickly, as it was not reproducible on my end.

The commands given were to allow you to verify that it was the correct fix in your circumstances now, without waiting on another beta release. The IDE will fix it itself if it has to.

So yes, either set the execute bit yourself if you need to get on with things now, or wait for the next IDE beta. Either way will do the same thing.
If you still have an issue, then please submit another ticket with a full bug report package so we have more context as to what is going on.

Thanks.

**Comment by Emc1923** [comment]
Verified fixed as of  IDE v2024.1100.0.687  Runtime v2024.1100.0.708.
